core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -303,6 +303,20 @@ public String getCarbonIndexFilePath(String taskId, String partitionId, String s
     }
   }
 
+  public String getCarbonIndexFilePath(String taskId, String partitionId, String segmentId,
+      int batchNo, String bucketNumber, String timeStamp,
+      ColumnarFormatVersion columnarFormatVersion) {
+    switch (columnarFormatVersion) {
+      case V1:
+      case V2:
+        return getCarbonIndexFilePath(taskId, partitionId, segmentId, bucketNumber);
+      default:
```

Comment:
V3 will go as default, this function is not created by myself, just added batchNo based on original one to get correct index path.